### PR TITLE
OpenCL things

### PR DESCRIPTION
**Pull request by oxij**
###### Motivation for this change

Side-effect of me doing OpenCL things.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/hwloc/default.nix [hunk 1]
```diff
@@ -24,7 +25,7 @@ stdenv.mkDerivation rec {
   # derivation and set optional dependencies to `null'.
   buildInputs = stdenv.lib.filter (x: x != null)
    ([ expat ncurses ]
-     ++  (optionals x11Support [ cairo libX11 ])
+     ++  (optionals x11Support [ libX11 cairo ])
```
Comment (FRidh) [inline pkgs/development/libraries/hwloc/default.nix]: accidental?

**Comment by oxij** [comment]
> accidental?

No, just OCD. `cairo` depends on `libX11` (via `mesa`), so I toposorted them :)

**Comment by FRidh** [comment]
Changes like that are just noise for reviewers, please don't do that.

**Comment by oxij** [comment]
Well, fine, removed.

But would you remove whitespace at the end of a line you didn't touch in a file you did touch in a PR? I would.
I consider this to be similar.

**Comment by fpletz** [comment]
@GrahamcOfBorg build ocl-icd

**Comment by oxij** [comment]
Thanks!

**Pull request merged**